#14 feat : 쇼핑몰 UI 구현, Toss위젯 연결

odukong opened the pull request:
## #️⃣ Issue Number

<!--- ex) #이슈번호, #이슈번호 -->
#14 

## 📝 요약(Summary)

<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
1. 쇼핑몰 UI를 구현하였습니다.
    - **쇼핑몰 메인 페이지, 상품 상세 페이지, 장바구니 페이지 / 결제 페이지, 결제결과 페이지**
2. 결제 관련한 Toss위젯을 연결하였습니다. 
    - 위젯 연결을 위해 `@tosspayments/payment-widget-sdk` 패키지를 설치하였습니다.

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [X] 새로운 기능 추가
- [ ] 버그 수정
- [X] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [X] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [X] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)
### 쇼핑몰 메인페이지
![image](https://github.com/user-attachments/assets/1570b7ef-3345-485f-a9a3-a0fc98c42b09)

### 쇼핑몰 상세페이지
![image](https://github.com/user-attachments/assets/2e9fa540-a79a-4b6d-92c7-c3e98938cd7a)

### 장바구니 및 즉시 결제 페이지
![image](https://github.com/user-attachments/assets/de4cd456-99fb-4020-9b6c-df5cc3fbcd9b)
![image](https://github.com/user-attachments/assets/ee45bbb3-79fd-45cf-a1b7-df790e404612)
![image](https://github.com/user-attachments/assets/1e10a4f2-2c61-4efe-90ed-a991533d38fd)

### (Toss) 결제 페이지
![image](https://github.com/user-attachments/assets/af491378-391d-4f1a-b064-33ff933a9ad1)
![image](https://github.com/user-attachments/assets/806afea8-92b9-4153-bb28-49d9869047af)


## 💬 공유사항 to 리뷰어

<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->
- 카드번호를 통한 할인율 적용로직은 아직 파악이 잘 안돼서.. 제가 임의로 구현해둔 상태입니다 💧

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [X] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

odukong committed:
feat : 쇼핑몰 메인 페이지 UI 구현

ref : #14

odukong committed:
feat : 상품 상세페이지 UI 구현

- 상품 상세페이지 구현
- 장바구니 cartStore 생성
  - 장바구니에는 아이템의 id, 아이템 수량, 가격을 저장하였으며, 장바구니 렌더링에는 id를 이용하여 상품 정보를 가져올 예정.
  - cartStore에 들어간 값과 비교하여 장바구니에 들어가있는 상품인지 구별.

ref : #14

odukong committed:
feat : 장바구니 결제 및 단독 결제 UI 구현

- 여러 품목을 결제하는 장바구니 결제 UI 구현
- 단일 품목을 결제하는 단독 결제 UI 구현
- 쇼핑몰 UI 에서 공통적으로 사용되는 ProductItem, CartItem 타입을 전역선언하여 코드리팩토링을 진행하였습니다.

ref : #14

odukong committed:
chore : toss widget 라이브러리 설치

ref : #14

odukong committed:
refactor : 상품 타입(productItem) 전역 타입 변경

ref : #14

odukong committed:
feat : 장바구니 결제 및 단독 결제 Toss 위젯 연결

- 장바구니 결제(다수품목)와 단독 즉시결제 로직을 구분하여 구현하였습니다.
- 공통적으로 결제하기 클릭 시, 결제 대기 상태 상품인 pendingStore로 저장되도록 구현하여 결제인증API 연결 시 정보를 전달할 수 있도록 하였습니다.
- 토스 UI + 토스 결과창 UI 구현

ref : #14

loading1031 commented:
굿굿 고생하셨습니답

카드 번호를 통한 혜택 적용 로직은
1. 서버에서 혜택이 적용되는 상품명을 제공
2. 상품명 str 매칭으로 적용

하는 형태로 러프하게 진행해야될거 같아요
저희 서버에 있는 카드 혜택과 쇼핑몰의 외부 DB를 외래키로 연결할 수 없기 때문에
판매 상품명을 문자열로 매칭하는게 유일한 방법일거 같습니다...허허

odukong committed:
refactor : 주문생성 API에 필요한 addToCartOrder 값추가

- 주문생성하기 API에 추가된 장바구니에 담긴 순서를 의미하는 addToCartOrder 키 추가 (단독 결제의 경우, 1로 고정됨)
- CartItem 타입에 addToCartOrder를 포함하도록 수정

ref : #14
